fix: semantic dependency upload issue

mittalVaghela opened the pull request:
<!-- 
Thanks for your contribution! 
Please follow the instructions on your PRs title and description.
aligned title description: '(feat|fix|chore|doc): _description of introduced change_'
Important: Contributing Guidelines can be found here: https://eclipse-tractusx.github.io/docs/oss/how-to-contribute
Info: <!- text comments ->  will be hidden from the rendered preview of your PR.
-->
## Description
Bug Fix:  fix semantic dependent model upload issue

### Following changes :
- Update SAMM SDK Version to 2.9.8
- Update Jena Server to 5.0.0 
- Update common-io version 2.17
- Update the validation logic according to new version
- Remove Unwanted code
- Added test case for samm dependent model validation 


## Issue:
Fixes #237


## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

agg3fe commented:
@mittalVaghela Thanks for the updates.
IMO, it makes more sense to update the samm sdk version to the latest. Here are latest versions released https://github.com/eclipse-esmf/esmf-sdk/releases
It is best when semantic models repository and semantic-hub have same version. Since semantic model repo (https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/generate.sh#L33) uses 2.9.7 version, so semantic-hub should also have the same version. 
Updating to samm sdk 2.8.1 will not be much beneficial as it will still have some issues with it.
FYI @tunacicek @jSchuetz88

mittalVaghela commented:
> @mittalVaghela Thanks for the updates. IMO, it makes more sense to update the samm sdk version to the latest. Here are latest versions released https://github.com/eclipse-esmf/esmf-sdk/releases It is best when semantic models repository and semantic-hub have same version. Since semantic model repo (https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/generate.sh#L33) uses 2.9.7 version, so semantic-hub should also have the same version. Updating to samm sdk 2.8.1 will not be much beneficial as it will still have some issues with it. FYI @tunacicek @jSchuetz88

@Sahin


> @mittalVaghela Thanks for the updates. IMO, it makes more sense to update the samm sdk version to the latest. Here are latest versions released https://github.com/eclipse-esmf/esmf-sdk/releases It is best when semantic models repository and semantic-hub have same version. Since semantic model repo (https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/generate.sh#L33) uses 2.9.7 version, so semantic-hub should also have the same version. Updating to samm sdk 2.8.1 will not be much beneficial as it will still have some issues with it. FYI @tunacicek @jSchuetz88

@agg3fe Thank you for your suggestion. I will update it to the latest version, 2.9.7, and let you know once it's ready for final review.

mittalVaghela commented:
> > @mittalVaghela Thanks for the updates. IMO, it makes more sense to update the samm sdk version to the latest. Here are latest versions released https://github.com/eclipse-esmf/esmf-sdk/releases It is best when semantic models repository and semantic-hub have same version. Since semantic model repo (https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/generate.sh#L33) uses 2.9.7 version, so semantic-hub should also have the same version. Updating to samm sdk 2.8.1 will not be much beneficial as it will still have some issues with it. FYI @tunacicek @jSchuetz88
> 
> @agg3fe Thank you for your suggestion. I will update it to the latest version, 2.9.7, and let you know once it's ready for final review.

@agg3fe  i have updated the SAMM SDK to the latest version of ESMF (2.9.7) and Jena (5.0.0). Could you please verify this again?

tunacicek commented on backend/src/main/java/org/eclipse/tractusx/semantics/hub/persistence/triplestore/SAMMSdk.java:
```diff
@@ -104,79 +91,65 @@ public AspectModelUrn getAspectUrn( final Model model ) {
-         final Resource resource = ResourceFactory.createResource( checkAndReplaceToBammPrefix(aspectModelUrn.getUrn().toASCIIString()));
-         final Model model = tripleStoreRequester.apply( checkAndReplaceToBammPrefix(aspectModelUrn.getUrn().toString()));
-
-         if(!isBamm()){
```
This check is removed- Do we not need to load the namespaces for bamm?

tunacicek commented on backend/src/test/java/org/eclipse/tractusx/semantics/hub/ModelsApiTest.java:
```diff
@@ -967,7 +968,9 @@ public void testModelStatusUpdateByURN() throws Exception {
    }
 
-   @Test
+	 //issue its fix into latest jena version 4.9.0 or above
```
Why are the following tests disabled?

mittalVaghela replied:
Actully latest version of jena and  samm sdk validator it will take it as warning not as error so i comment out this test case

tunacicek commented on backend/src/main/java/org/eclipse/tractusx/semantics/hub/samm/SDKAccessHelperSAMM.java:
```diff
@@ -24,189 +24,135 @@
-         return Try.failure( e );
+       String modelData = persistenceLayer.getModelDefinition(AspectModelUrn.fromUrn(urn));
+       final AspectModel aspectModel =loadAspectModel(modelData).get();
+       final JsonSchemaGenerationConfig config = JsonSchemaGenerationConfigBuilder.builder()
```
Does it make sense to create this object one time on class init?

tunacicek commented on backend/src/main/java/org/eclipse/tractusx/semantics/hub/samm/SDKAccessHelperSAMM.java:
```diff
@@ -24,189 +24,135 @@
-   public String getOpenApiDefinitionJson( String urn, String baseUrl ) {
-      Aspect aspect = getBamAspect( urn );
-      AspectModelOpenApiGenerator openApiGenerator = new AspectModelOpenApiGenerator();
+			final DocumentationGenerationConfig config = DocumentationGenerationConfigBuilder.builder()
```
Also here: 
Does it make sense to create this object one time on class init?

mittalVaghela replied:
config object has single use only for this method i prefer to create this object when it will required.

tunacicek commented:
@mittalVaghela : thanks for the PR. 
Also the https://github.com/eclipse-tractusx/sldt-semantic-hub/blob/main/README.md needs to be adjusted to describe jena 5.0.0

tunacicek commented:
@mittalVaghela : You need to accept the eca -> https://api.eclipse.org/git/eca/status/gh/eclipse-tractusx/sldt-semantic-hub/309

mittalVaghela commented on backend/src/main/java/org/eclipse/tractusx/semantics/hub/TripleStoreConfiguration.java:
```diff
@@ -67,7 +67,6 @@ public FusekiServer fusekiServer( final TripleStoreProperties properties ) {
                          .verbose( true )
                          .contextPath( embedded.getContextPath() )
                          .enableStats( true )
-                         .enableCors( true )
```
ok i will revert this changes.

mittalVaghela commented on backend/src/main/java/org/eclipse/tractusx/semantics/hub/SDKAccessHelper.java:
```diff
@@ -67,11 +59,11 @@ public Try<String> getExamplePayloadJson( String urn  ) {
-   public Try<VersionedModel> loadBammModel( String modelString ) {
-      return sdkAccessHelperSAMM.loadSammModel( modelString );
-   }
+	public Try<AspectModel> loadAspectModel(String modelUrn){
```
project inbuild

agg3fe commented:
Hi @mittalVaghela, I just noticed one thing. The spring boot version in your local application is 3.2.4, but the latest spring boot version in tractusx repo is 3.2.5. There are some other dependencies also which are different in both pom.xml files. Could you please compare and update this in your repo and build the project again. Otherwise this might create some discrepancies in final build.
also this is important to create an IP issue for the new Restricted dependencies. That part I will handle as soon as pom.xml is updated.
Thanks

mittalVaghela commented:
> Hi @mittalVaghela, I just noticed one thing. The spring boot version in your local application is 3.2.4, but the latest spring boot version in tractusx repo is 3.2.5. There are some other dependencies also which are different in both pom.xml files. Could you please compare and update this in your repo and build the project again. Otherwise this might create some discrepancies in final build. also this is important to create an IP issue for the new Restricted dependencies. That part I will handle as soon as pom.xml is updated. Thanks

sure i will

agg3fe commented:
> @mittalVaghela IP issue for code content has been created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/19064 This is important since this PR contains more than 1k lines of code.

Currently we are waiting for the approval of IP issue raised.

mittalVaghela commented:
> > @mittalVaghela IP issue for code content has been created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/19064 This is important since this PR contains more than 1k lines of code.
> 
> Currently we are waiting for the approval of IP issue raised.

ok. but i updated [chart](version: "0.4.3", path: "charts/semantic-hub"))

mittalVaghela commented:
> > @mittalVaghela IP issue for code content has been created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/19064 This is important since this PR contains more than 1k lines of code.
> 
> Currently we are waiting for the approval of IP issue raised.

ok.

agg3fe commented on charts/semantic-hub/Chart.yaml:
```diff
@@ -26,8 +26,8 @@ sources:
-version: 0.4.2
-appVersion: 0.5.2
+version: 0.4.3
+appVersion: 0.5.3
```
for now, remove changes to `appVersion` and keep only `version`. We will need to build a new version 0.5.3 after this PR is merged. will create a new PR for that.
Also please add changelog to this PR, stating the major changes this PR has.

tunacicek commented on backend/src/main/java/org/eclipse/tractusx/semantics/hub/SDKAccessHelper.java:
```diff
@@ -67,11 +59,11 @@ public Try<String> getExamplePayloadJson( String urn  ) {
-   public Try<VersionedModel> loadBammModel( String modelString ) {
-      return sdkAccessHelperSAMM.loadSammModel( modelString );
-   }
+	public Try<AspectModel> loadAspectModel( String modelUrn ){
```
Code Format not correct.

tunacicek commented on backend/src/main/java/org/eclipse/tractusx/semantics/hub/persistence/triplestore/TripleStorePersistence.java:
```diff
@@ -107,6 +107,7 @@ public SemanticModel getModel( final AspectModelUrn urn ) {
    @Override
    public SemanticModel updateModel( String urn, SemanticModelStatus status ) {
 
+	  validateStatusParameter(status);
```
code format not correct

tunacicek commented on backend/src/test/java/org/eclipse/tractusx/semantics/hub/ModelsApiTest.java:
```diff
@@ -648,9 +648,10 @@ public void testSaveModelWithExternalReferencesExpectSuccessForBAMM() throws Exc
-            .andExpect( jsonPath( "$.error.details.ERR_PROCESSING",
-                  containsString( "urn:bamm:org.eclipse.tractusx.traceability:0.1.1#PartDataCharacteristic not found" ) ) );
-      // save the traceability aspect model
+				    .andExpect( jsonPath( "$.error.details.validationError",
```
format?

agg3fe commented:
> > @mittalVaghela IP issue for code content has been created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/19064 This is important since this PR contains more than 1k lines of code.
> 
> Currently we are waiting for the approval of IP issue raised.

@mittalVaghela IP issue is closed now. As soon as you fix the comments, we can merge the PR.
Also please squash the commits before we can merge it.

mittalVaghela commented:
> > > @mittalVaghela IP issue for code content has been created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/19064 This is important since this PR contains more than 1k lines of code.
> > 
> > 
> > Currently we are waiting for the approval of IP issue raised.
> 
> @mittalVaghela IP issue is closed now. As soon as you fix the comments, we can merge the PR. Also please squash the commits before we can merge it.

@agg3fe & @tunacicek , I have addressed all the comments and implemented the suggested changes. Please review it again and let me know if any further changes are needed. Thank you!

mittalVaghela reviewed (commented):
All changes its done

agg3fe reviewed (approved):
LGTM
